KenticoOnboardingApplication/src/KenticoOnboardingApplication.Contracts/Services/Wrappers/RetrievedItem.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace KenticoOnboardingApplication.Contracts.Services.Wrappers
+{
+    public class RetrievedItem<TItem> where TItem : class
+    {
+        public TItem Item { get; }
+        public bool WasFound => Item != null;
+        public static RetrievedItem<TItem> Empty = new RetrievedItem<TItem>();
+
+        private RetrievedItem() { }
+
+        public RetrievedItem(TItem item) =>
+            Item = item ?? throw new ArgumentNullException(nameof(item), "Null passed to RetrievedItem constructor.");
+
+        public override string ToString() => Item + (WasFound ? " was found" : " was not found");
```

Comment:
I love how this assembles to a sentence 😃 
```suggestion
        public override string ToString() => Item + (WasFound ? " and was found" : " Item was not found");
```